pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -53,39 +69,72 @@ func NewNode(constraints *v1alpha5.Constraints, instanceTypeOptions []cloudprovi
 }
 
 func (n Node) Compatible(pod *v1.Pod) error {
-	if n.unschedulable {
-		return errors.New("node is unschedulable")
+	podRequirements := v1alpha5.NewPodRequirements(pod)
+	if err := n.Requirements.Compatible(podRequirements); err != nil {
+		return err
 	}
 
-	podRequirements := v1alpha5.NewPodRequirements(pod)
-	if err := n.Constraints.Requirements.Compatible(podRequirements); err != nil {
+	// Taints should only arrive via the provisioner, so this should never occur but to be safe
```

Comment:
I'm not sure we want to "be safe" in a tight loop where performance matters. Is there something we can do to protect us against this theoretical case?

Reply:
My understanding is that do *do* need to check the tolerations somewhere. Is it happening somewhere else?

Reply:
The test "should schedule pods that tolerate provisioner constraints" checks for this and the toleration check currently happens during selection. Pods never make it this far since they fail to pass selection.  Removing it for now, selection may be reworked in the future but there's a test for it.